pkg/ebpf/c/bpf_telemetry.h

```diff
@@ -5,69 +5,62 @@
 #include "telemetry_types.h"
 #include "map-defs.h"
 
+// this macro reads the stack slot at offset 512
+// If everything went okay, then it should have the pointer
+// to the telemetry map.
+#define FETCH_TELEMETRY_BLOB() ({                   \
+    instrumentation_blob_t *__tb;                   \
+    asm("%0 = *(u64 *)(r10 - 512)" : "=r"(__tb));   \
+    __tb;                                           \
+})
+
+BPF_ARRAY_MAP(bpf_instrumentation_map, instrumentation_blob_t, 1);
+
+#ifdef INSTRUMENTATION_ENABLED
```

Comment:
suggestion: define flags usually don't include the word "enabled"
```suggestion
#ifdef HELPER_INSTRUMENTATION
```